optimize docs for event system (#1903)
cocos-creator/creator-docs: en/engine/event/event-emit.md

# Event Listening and Emitting

Cocos Creator provides the `EventTarget` class to suport the listening and emitting of custom event. Before using it, it needs to be imported from the `'cc'` module and an `EventTarget` object needs to be instantiated.
```ts
import { EventTarget } from 'cc';
const eventTarget = new EventTarget();
```

> > __Note__: Although the `Node` object also implements some `EventTarget` interfaces, we no longer recommend continuing to use the `Node` object to listen to and emit custom events. Because this is not efficient enough, and we also hope that `Node` The object only cares about events related to `Node`.

Reviewer: ```suggestion
> __Note__: although the `Node` object also implements some `EventTarget` interfaces, it is no longer recommended to continue using the `Node` object to listen to and emit custom events. Because this is not efficient enough, and we also hope that `Node` The object only cares about events related to `Node`.
```

Reviewer: But this statement:
```
Because this is not efficient enough, and we also hope that `Node` The object only cares about events related to `Node`.
```

doesn't make much sense to me, can you explain what you meant?

Author: Sorry, I updated this sentence https://github.com/cocos-creator/creator-docs/pull/1903/commits/ff275529711a93dc93ff7d78fe457106804eee07

What I want to emphasize is that we don't hope developers to coding like
```
this.node.on('foobar', callback, target);  // the event 'foobar' is not node-related

eventTarget.on('foobar', callback, target);  // we suggest to register 'foobar' on eventTarget, instead of node
```

## Event Listening

Listen to events can be registered by the interface `this.node.on()`. The methods are as follows:

```ts
// This event monitor is triggered every time and needs to be unregistered manually.
eventTarget.on(type, func, target?);
```

The `type` is the event registration string. `func` is the callback to listen to when the event is executed. And `target` is the event receive object. If `target` is not set, then `this` in the callback refers to the object that is currently executing the callback.

> The event listener interface `on` can pass to the third parameter target to bind the caller of the response function. The following two calling methods have the same effect:

Reviewer: ```suggestion
The event listener interface `on` can pass to the third parameter target to bind the caller of the response function. Calling the following two methods would have the same effect:
```

```ts
// Using Function Binding
eventTarget.on(Node.EventType.MOUSE_DOWN, function ( event ) {
  this.enabled = false;
}.bind(this));

// Using the third parameter
eventTarget.on(Node.EventType.MOUSE_DOWN, (event) => {
  this.enabled = false;
}, this);
```

> Besides listening with `on`, we can also use the `once` interface. The `once` listener will shut the event being listened to after the listener function responds.

Reviewer: ```suggestion
Besides listening with `on`, the `once` interface can also be used. The `once` listener will shut the event being listened to after the listener function responds.
```

}
```

## Event Emitting

> We can emit events through the `eventTarget.emit()` interface, as follows:

Reviewer: ```suggestion
Events are emitted through the `eventTarget.emit()` interface, as follows:
```
